<a href="https://colab.research.google.com/github/ryry91021/Blackjack-Game/blob/main/Pandas_hw4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Please enter your name and student ID here.**

In [ ]:
#Name: Ryan Davis
#Student ID: 20010256
#Pledge: I pledge my Honor that I have abided by the Stevens Honor System- RD

**Homework - Pandas**

**Use Pandas to solve questions 1-5.**

In [ ]:
# Mount your drive
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
# Change current directory
%cd "/content/drive/MyDrive"

/content/drive/MyDrive


In [ ]:
 # Verify the current directory
%pwd

'/content/drive/MyDrive'

In [ ]:
 # Verify the current directory
%ls

'10 thuings I hate abou twhitman.gdoc'
'12-12 schedule.gsheet'
 416-283-828_3pm9_147.jpg
 416-283-828_43bq_167.jpg
 416-283-828_4bc3_143.jpg
 416-283-828_6mg8_170.jpg
 416-283-828_6tcq_150.jpg
 416-283-828_7849_146.jpg
 416-283-828_7mr2_142.jpg
 416-283-828_7qy2_166.jpg
 416-283-828_q867_169.jpg
 416-283-828_qhg4_145.jpg
 416-283-828_qyc6_149.jpg
 416-283-828_x7k4_172.jpg
 416-283-828_xgt8_168.jpg
 416-283-828_xqqz_148.jpg
 416-283-828_y43q_144.jpg
'bIG CHIP GUYS (1).mp4'
'bIG CHIP GUYS.mp4'
'cal 103 notes.gdoc'
'CAL Research paper first draft.gdoc'
'Caminante proyecto.mp4'
'Captar El Momento.mp4'
'Colab Notebooks'/
'COLLEGE ESSAY.gdoc'
'Copy of Analytical Essay Final Copy.gdoc'
'Copy of Chem Project rough draft.gdoc'
'Copy of Copy of Research essay final draft (1).gdoc'
'Copy of Copy of Research essay final draft.gdoc'
'Cover Letter.gdoc'
'Creative Writing Final.gdoc'
 Dialogue.gdoc
'Done famil.gsheet'
'dorm questions.gdoc'
'Driveway- Ryan Davis.gdoc'
'Dr. Muller.gdoc'
'El Decimo.mov'

## Q1. (0.5 points)
Import the Pandas package and load the data from **EmissionsData.csv** into a dataframe called **em_df**.

In [ ]:
# Provide your answer here.
import pandas as pd
em_df=pd.read_csv("EmissionsData.csv")

## Q2. (1 point)

Sort the dataframe using the `sort_values()` function by the column '**Emission**' in descending order. Display the top 5 records.

<div style="overflow-x:auto;">
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Year</th>
      <th>Country</th>
      <th>Continent</th>
      <th>Emission</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>592</th>
      <td>2008</td>
      <td>Qatar</td>
      <td>Asia</td>
      <td>48.601620</td>
    </tr>
    <tr>
      <th>593</th>
      <td>2009</td>
      <td>Qatar</td>
      <td>Asia</td>
      <td>44.836401</td>
    </tr>
    <tr>
      <th>595</th>
      <td>2011</td>
      <td>Qatar</td>
      <td>Asia</td>
      <td>44.018926</td>
    </tr>
    <tr>
      <th>594</th>
      <td>2010</td>
      <td>Qatar</td>
      <td>Asia</td>
      <td>42.639076</td>
    </tr>
    <tr>
      <th>710</th>
      <td>2010</td>
      <td>Trinidad And Tobago</td>
      <td>North America</td>
      <td>38.337841</td>
    </tr>
  </tbody>
</table>
</div>

In [ ]:
# Provide your answer here.
em_df.sort_values(by='Emission', ascending=False).head(5)

Year              Country      Continent   Emission
592  2008                Qatar           Asia  48.601620
593  2009                Qatar           Asia  44.836401
595  2011                Qatar           Asia  44.018926
594  2010                Qatar           Asia  42.639076
710  2010  Trinidad And Tobago  North America  38.337841

## Q3.

### Q3.1 (1 point)

Use the `groupby()` function to group **em_df** by the '**Year**' column..

In [ ]:
# Provide your answer here.
em_df=em_df.groupby('Year')

### Q3.2 (1.5 points)

Retrieve the group '**2011**' using the `get_group()` function, and sort the 2011 group dataframe by '**Emission**' in ascending order. Display the first 5 rows.


In [ ]:
# Provide your answer here.
em_df.get_group(2011).sort_values(by='Emission').head(5)

Year           Country Continent  Emission
51   2011           Burundi    Africa  0.021350
683  2011              Chad    Africa  0.043830
779  2011  Dem Rep Of Congo    Africa  0.050303
639  2011           Somalia    Africa  0.058707
607  2011            Rwanda    Africa  0.062874

### Q3.3 (1 point)

Use the `mean()` function to calculate the average emission of all countries in the year **2011**.

<div style="overflow-x:auto;">
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Year</th>
      <th>Country</th>
      <th>Continent</th>
      <th>Emission</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>51</th>
      <td>2011</td>
      <td>Burundi</td>
      <td>Africa</td>
      <td>0.021350</td>
    </tr>
    <tr>
      <th>683</th>
      <td>2011</td>
      <td>Chad</td>
      <td>Africa</td>
      <td>0.043830</td>
    </tr>
    <tr>
      <th>779</th>
      <td>2011</td>
      <td>Dem Rep Of Congo</td>
      <td>Africa</td>
      <td>0.050303</td>
    </tr>
    <tr>
      <th>639</th>
      <td>2011</td>
      <td>Somalia</td>
      <td>Africa</td>
      <td>0.058707</td>
    </tr>
    <tr>
      <th>607</th>
      <td>2011</td>
      <td>Rwanda</td>
      <td>Africa</td>
      <td>0.062874</td>
    </tr>
  </tbody>
</table>
</div>

```text
Average: 4.904900274527919
```

In [ ]:
# Provide your answer here.
em_df.get_group(2011)["Emission"].mean()

4.904900274527918

## Q4.

### Q4.1 (2.0 points)

Find the rows in `em_df` that meet the following conditions: **Year is 2010, Continent is Europe, and Emission is greater than 2**. Sort the resulting dataframe by '**Emission**' in descending order, and display the first 5 rows.


In [ ]:
# Provide your answer here.
em_df.get_group(2010)[(em_df.get_group(2010)["Continent"]=="Europe") & (em_df.get_group(2010)["Emission"]>2)].sort_values(by='Emission', ascending=False)

Year                 Country Continent   Emission
434  2010              Luxembourg    Europe  21.635136
226  2010                 Estonia    Europe  13.773197
246  2010           Faroe Islands    Europe  12.986678
534  2010                  Norway    Europe  11.616193
234  2010                 Finland    Europe  11.530840
530  2010             Netherlands    Europe  10.949316
182  2010          Czech Republic    Europe  10.613368
54   2010                 Belgium    Europe  10.012610
186  2010                 Germany    Europe   9.179817
334  2010                 Ireland    Europe   8.810151
198  2010                 Denmark    Europe   8.394658
574  2010                  Poland    Europe   8.334872
42   2010                 Austria    Europe   8.127693
258  2010          United Kingdom    Europe   7.841653
286  2010                  Greece    Europe   7.797370
658  2010                Slovenia    Europe   7.530605
178  2010                  Cyprus    Europe   6.983908
354  2010                   Italy    Europe   6.838375
654  2010         Slovak Republic    Europe   6.721960
730  2010                 Ukraine    Europe   6.638152
86   2010                 Belarus    Europe   6.556550
474  2010                   Malta    Europe   6.316496
642  2010                  Serbia    Europe   6.292017
346  2010                 Iceland    Europe   6.168529
6    2010                 Andorra    Europe   6.124770
70   2010                Bulgaria    Europe   5.966884
222  2010                   Spain    Europe   5.815787
662  2010                  Sweden    Europe   5.606000
82   2010  Bosnia And Herzegovina    Europe   5.513994
242  2010                  France    Europe   5.496707
322  2010                 Hungary    Europe   5.074750
130  2010             Switzerland    Europe   4.952499
314  2010                 Croatia    Europe   4.728822
582  2010                Portugal    Europe   4.690095
430  2010               Lithuania    Europe   4.319018
466  2010               Macedonia    Europe   4.171161
482  2010              Montenegro    Europe   4.167664
438  2010                  Latvia    Europe   3.963228
598  2010                 Romania    Europe   3.918410
46   2010              Azerbaijan    Europe   3.388226

### Q4.2 (1 point)

Calculate the average emission of **the resulting dataframe from the previous conditions**.

<div style="overflow-x:auto;">
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Year</th>
      <th>Country</th>
      <th>Continent</th>
      <th>Emission</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>434</th>
      <td>2010</td>
      <td>Luxembourg</td>
      <td>Europe</td>
      <td>21.635136</td>
    </tr>
    <tr>
      <th>226</th>
      <td>2010</td>
      <td>Estonia</td>
      <td>Europe</td>
      <td>13.773197</td>
    </tr>
    <tr>
      <th>246</th>
      <td>2010</td>
      <td>Faroe Islands</td>
      <td>Europe</td>
      <td>12.986678</td>
    </tr>
    <tr>
      <th>534</th>
      <td>2010</td>
      <td>Norway</td>
      <td>Europe</td>
      <td>11.616193</td>
    </tr>
    <tr>
      <th>234</th>
      <td>2010</td>
      <td>Finland</td>
      <td>Europe</td>
      <td>11.530840</td>
    </tr>
  </tbody>
</table>
</div>

```text
Average: 7.4887038956
```

In [ ]:
# Provide your answer here.
em_df.get_group(2010)[(em_df.get_group(2010)["Continent"]=="Europe") & (em_df.get_group(2010)["Emission"]>2)].sort_values(by='Emission', ascending=False)["Emission"].mean()

7.4887038956

## Q5.

In [ ]:
em_df=pd.read_csv("EmissionsData.csv")
type(em_df)


pandas.core.frame.DataFrame

### Q5.1 (1 point)

In **em_df**, create a new column called '**Emission10PMore**'. The value in this column should be 10% more than the corresponding value in the original '**Emission**' column.


In [ ]:
# Provide your answer here.
em_df["Emission10PMore"]=em_df["Emission"]*1.1

### Q5.2 (1 point)

Sort the resulting dataframe by '**Emission10PMore**' in ascending order and display the first 5 rows.

<div style="overflow-x:auto;">
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Year</th>
      <th>Country</th>
      <th>Continent</th>
      <th>Emission</th>
      <th>Emission10PMore</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>50</th>
      <td>2010</td>
      <td>Burundi</td>
      <td>Africa</td>
      <td>0.020542</td>
      <td>0.022596</td>
    </tr>
    <tr>
      <th>49</th>
      <td>2009</td>
      <td>Burundi</td>
      <td>Africa</td>
      <td>0.020868</td>
      <td>0.022954</td>
    </tr>
    <tr>
      <th>51</th>
      <td>2011</td>
      <td>Burundi</td>
      <td>Africa</td>
      <td>0.021350</td>
      <td>0.023485</td>
    </tr>
    <tr>
      <th>48</th>
      <td>2008</td>
      <td>Burundi</td>
      <td>Africa</td>
      <td>0.021615</td>
      <td>0.023777</td>
    </tr>
    <tr>
      <th>681</th>
      <td>2009</td>
      <td>Chad</td>
      <td>Africa</td>
      <td>0.042689</td>
      <td>0.046958</td>
    </tr>
  </tbody>
</table>
</div>


In [ ]:
# Provide your answer here.
em_df.sort_values(by='Emission10PMore').head(5)

Year  Country Continent  Emission  Emission10PMore
50   2010  Burundi    Africa  0.020542         0.022596
49   2009  Burundi    Africa  0.020868         0.022954
51   2011  Burundi    Africa  0.021350         0.023485
48   2008  Burundi    Africa  0.021615         0.023777
681  2009     Chad    Africa  0.042689         0.046958